################################################################################
SHORT DESCRIPTION: 
################################################################################
Add the Mosart 7105 MT multitouch input device (Panasonic CF-C1 systems).

################################################################################
LONG DESCRIPTION: 
################################################################################
Add the Mosart 7105 MT multitouch input device to support Panasonic CF-C1
systems in hid-ids.h list.

################################################################################
CHANGELOG 
################################################################################
Original Author: Ross Philipson <philipsonr@ainfosec.com>
Port to 3.18: Eric Chanudet <chanudete@ainfosec.com>
Port to 4.14: Richard Turner <turnerr@ainfosec.com>

################################################################################
REMOVAL 
################################################################################
Depending on the HCL, MOsart 7105 MT multitouch input device is supported or
not.

################################################################################
UPSTREAM PLAN
################################################################################
None.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
None.

################################################################################
PATCHES 
################################################################################
--- a/drivers/hid/hid-ids.h
+++ b/drivers/hid/hid-ids.h
@@ -1054,6 +1054,7 @@
 #define USB_DEVICE_ID_TURBOX_KEYBOARD	0x0201
 #define USB_DEVICE_ID_ASUS_MD_5110	0x5110
 #define USB_DEVICE_ID_TURBOX_TOUCHSCREEN_MOSART	0x7100
+#define USB_DEVICE_ID_TURBOX_TOUCHSCREEN_MOSART5 0x7105
 
 #define USB_VENDOR_ID_TWINHAN		0x6253
 #define USB_DEVICE_ID_TWINHAN_IR_REMOTE	0x0100
--- a/drivers/hid/hid-multitouch.c
+++ b/drivers/hid/hid-multitouch.c
@@ -1618,6 +1618,9 @@ static const struct hid_device_id mt_dev
 	{ .driver_data = MT_CLS_CONFIDENCE_MINUS_ONE,
 		MT_USB_DEVICE(USB_VENDOR_ID_TURBOX,
 			USB_DEVICE_ID_TURBOX_TOUCHSCREEN_MOSART) },
+	{ .driver_data = MT_CLS_CONFIDENCE_MINUS_ONE,
+		HID_USB_DEVICE(USB_VENDOR_ID_TURBOX,
+			USB_DEVICE_ID_TURBOX_TOUCHSCREEN_MOSART5) },
 
 	/* Panasonic panels */
 	{ .driver_data = MT_CLS_PANASONIC,
